pyccel/codegen/printing/fcode.py

```diff
@@ -452,28 +455,151 @@ def _print_Import(self, expr):
+                longest_chunk = length
+                cuts = []
+                positions = [idx for idx in range(length) if line[idx] == ' ']
+                if len(positions) > 12:
+                    positions = positions[int(len(positions) / 4) : 12]
+                for mask in range(1, 1<<len(positions)):
+                    mask_cuts = [positions[idx] for idx in range(len(positions)) if (1 << idx)&mask != 0 ]
+                    mask_cuts.append(length)
+                    valid = True
+                    mask_longest_chunk = 0
+                    for idx in range(len(mask_cuts) - 1):
+                        if mask_cuts[idx + 1] - mask_cuts[idx] < 8:
+                            valid = False
+                        mask_longest_chunk = max(mask_longest_chunk, mask_cuts[idx + 1] - mask_cuts[idx])
+                    if valid and mask_longest_chunk < longest_chunk:
+                        cuts = mask_cuts
+                        longest_chunk = mask_longest_chunk
+                if longest_chunk != length:
+                    return [line[a + 1 : b + 1] for a, b in zip([-1, *cuts], cuts)]
+                return [line[i:i + int(length / 3 + 1)] for i in range(0, length, int(length / 3 + 1))]
+
+            if fargs_format == ['*']:
+                return ', '.join(['print *', *fargs]) + '\n'
+            separator = self._print_LiteralString(LiteralString(sep))
+            args_formatting = ''
+            args_code = ''
+            for a_f, a_c in zip(fargs_format, fargs):
+                if a_f == 'A' and len(a_c) > 30 and a_c[0] == '\'' and a_c.count('\'') == 2:
+                    chunks = split_long_argument(a_c)
+                    whole_format = ' '.join(['A'] * len(chunks))
+                    arg_concat = ' , '.join('\'{}\''.format(chunk) for chunk in chunks)
+                    args_code += ('' if args_formatting == ''
+                                  else (' , ' + separator + ' , ' if separator != '' else ' , ')) + arg_concat
+                    args_formatting += ('' if args_formatting == ''
+                                        else (' A ' if separator != '' else ' ')) + whole_format
+                else:
+                    args_code += ('' if args_formatting == ''
+                                  else (' , ' + separator + ' , ' if separator != '' else ' , ')) + (a_c if a_c != '' else '\'\'')
+                    args_formatting += ('' if args_formatting == ''
+                                        else (' A ' if separator != '' else ' ')) + a_f
+            if args_formatting == '':
+                return "write(*, '(A)', advance=\"no\") {}\n".format(self._print_LiteralString(LiteralString(end)))
+            if end != '':
+                return "write(*, '({} A)', advance=\"no\") {} , {}\n".format(args_formatting, args_code, self._print_LiteralString(LiteralString(end)))
+            return "write(*, '({})', advance=\"no\") {}\n".format(args_formatting, args_code)
```

Comment:
I don't see a line with `advance=\"yes\"`. Have I missed it? Or are the new lines handled with the `end` parameter? I think it would be better to use `advance=\"yes\"` to ensure that the output is flushed

Reply:
oh thank you for mentioning that! So flushing should be performed only when the end is not specified as kwarg or whenever end contains a '\n'? I did not need to use  advance=\"yes\" because i format the ending myself and add it as an extra argument of write.

Reply:
Hmm, that's true. Actually it seems that python's `print` has a `flush` argument which defaults to `False` so maybe it is better like this